compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1546,6 +1549,16 @@ public void visit(BLangReturn astReturnStmt) {
         }
     }
 
+    private BLangDiagnosticLocation getFunctionLastLinePos() {
+        if (this.env.enclFunc.pos == null) {
+            return null;
```

Comment:
Lets create an issue for fix null position in functions?

Reply:
This happens only for generated object `$init$` function. 
Added from https://github.com/ballerina-platform/ballerina-lang/pull/35250
If we pass a position for this, it will be shown in the stackFrame.